.github/PULL_REQUEST_TEMPLATE/pull_request_template.md

```diff
@@ -0,0 +1,16 @@
+# Pull Request Checklist
+
+Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:
+
+- [ ] Does the API design and implementation within your PR follow the [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)?
+  - [ ] Do you have an issue or spec doc that describes the design, usage, and motivating scenario?
+- [ ] If you are modifying header files, are the changes documented using doxygen style comments?
+- [ ] Have you added relevant unit tests to ensure CI will catch future regressions?
+- [ ] Have you reviewed your own PR to make sure there aren't any unintended changes or commits?
+- [ ] Does your PR have a descriptive title and description (with issue number), which explains why the change is being made?
+  - [ ] Does your PR have a single purpose or does it have other unrelated changes that can be separated out into its own PR?
+- [ ] Do you have any complex components that could use comments in source to explain (focusing on the "why")?
+- [ ] Are there any typos or spelling errors, especially in user-facing documentation?
+- [ ] Is the PR actually ready for review or is it work-in-progress (WIP) or an experiment? If so, can it start off as a draft PR?
+- [ ] Do your changes have impact elsewhere? For instance, do you need to update other docs or exiting markdown files that might be impacted?
+- [ ] Does the PR contain any breaking changes?
```

Comment:
I think this is awesome!
I wonder if we could make it faster by adding just a tittle for each bullet and the adding the description for each one below. Expecting someone could go quickly to each bullet and only go to the entire description if it is someone new or there's something recently added to the list.
The motivation for this would be making this quick labels that can be mentally learned with each PR.  A mental label is easier to reference and learn than an entire paragraph.

Something like
 - [ ] Guideline?
  - [ ] Related issue?
- [ ] Doxygen
- [ ] Unit test
- [ ] unwanted commits
- [ ] descriptive tittle
  - [ ] single purpose
- [ ] Source comments
- [ ] Typos?
- [ ] is WIP?
- [ ] external refs?
- [ ] breaking changes?

then, we can have the same list but with larger description about what each label means. Here we can also add help about how to check this label better, like what tools we can use to find typos, or how to find non-utf chars? , We can basically try to enhance each label to set a quality bar and help people and ourself to reach that bar.
Also, when reviewing a PR, we can then just mention the label of what is wrong and then link to this reference guide to understand about that label.

Let me know what you think